SEBrowser/Controllers/OpenXDA/TrendChannelController.cs

```diff
@@ -0,0 +1,353 @@
+                    {(string.IsNullOrEmpty(assetFilter) ? "" : $"AND {assetFilter}")}";
+
+                string query =
+                    $@"SELECT
+	                    Channel.ID,
+	                    Channel.Name,
+	                    Channel.Description,
+                        Asset.ID as AssetID,
+	                    Asset.AssetKey,
+	                    Asset.AssetName,
+                        Meter.ID as MeterID,
+	                    Meter.AssetKey AS MeterKey,
+	                    Meter.Name AS MeterName,
+                        Meter.ShortName AS MeterShortName,
+	                    Phase.Name AS Phase,
+	                    ChannelGroup.Name AS ChannelGroup,
+	                    ChannelGroupType.DisplayName AS ChannelGroupType,
+	                    ChannelGroupType.Unit
+                    FROM 
+	                    Channel LEFT JOIN
+	                    Phase ON Channel.PhaseID = Phase.ID LEFT JOIN
+	                    Asset ON Asset.ID = Channel.AssetID LEFT JOIN
+	                    Meter ON Meter.ID = Channel.MeterID LEFT JOIN
+	                    ChannelGroupType ON Channel.MeasurementCharacteristicID = ChannelGroupType.MeasurementCharacteristicID AND Channel.MeasurementTypeID = ChannelGroupType.MeasurementTypeID LEFT JOIN
+	                    ChannelGroup ON ChannelGroup.ID = ChannelGroupType.ChannelGroupID
+                    WHERE
+	                    {filters}";
+
+                DataTable table = connection.RetrieveData(query);
+
+                return table;
+            }
+        }
+
+        [Route("GetLineChartData"), HttpPost]
+        public IHttpActionResult GetLineChartData([FromBody] JObject postData)
+        {
+            Task<Stream> streamTask = XDAAPIHelper.Post("api/HIDS/QueryPoints", new StringContent(postData.ToString(), Encoding.UTF8, "application/json"));
+            IEnumerable<int> channelIds = postData["Channels"].ToObject<IEnumerable<int>>();
+            List<ChannelDataSet> channelList = channelIds.Select(id => new ChannelDataSet()
+            {
+                Points = new List<object>(),
+                ChannelID = id.ToString("x8")
+            }).ToList();
+            using (Stream stream = streamTask.Result)
+            using (TextReader reader = new StreamReader(stream))
+            {
+                while (true)
+                {
+                    string line = reader.ReadLine();
+
+                    if (line == null)
+                        break;
+
+                    if (line == string.Empty)
+                        continue;
+
+                    LinearPoint point = JObject
+                        .Parse(line)
+                        .ToObject<LinearPoint>();
+
+                    //TODO: Perform downsampling to get to something reasonable for a webpage here (maybe 100 or so points per channel?). This would ignore a point before insertion
+                    int index = channelList.FindIndex(chan => chan.ChannelID == point.Tag);
```

Comment:
it would be faster to use a `Dictionary` or `Hashset` rather than a `List`. Is there a reason it needs to be a `List`?

Reply:
Yep, changed